### ISSUE TITLE
require-error false positives

### ISSUE THREAD

**Issue by jalaziz** [Antonboom/testifylint#62]
`require-error` currently warns for the following pattern:

```
	for _, tc := range testCases {
		t.Run(tc.testName, func(t *testing.T) {
			// ... some logic
			if tc.someValue == nil && tc.someOtherValue == nil {
				assert.Nil(t, result)
				assert.NoError(t, err)  // <--- testifylint complains here
			} else if tc.someOtherValue != nil {
				assert.EqualError(t, err, tc.expectedError.Error())
			} else {
				assert.Equal(t, tc.expectedError, err)
				assert.Equal(t, tc.expectedValue, result)
			}
		})
```

Given that there is no logic after the final assert in the block and no logic after the conditional block, it seems like `require` should not strictly be required.

**Comment by Antonboom** [comment]
@jalaziz hi!

what version of `testifylint` / `golangci-lint` are you using?

**Comment by jalaziz** [comment]
I'm using `golangci-lint` v1.56.0 which contains `testifylint` 1.1.0.

**Comment by Antonboom** [comment]
@jalaziz hi!

could you please run on your codebase (to check fixes):
```go
go install github.com/Antonboom/testifylint@latest
testifylint --disable-all --enable=expected-actual,require-error ./...
```

**Comment by jalaziz** [comment]
@Antonboom 

Can confirm the `expected-actual` and `require-error` fixes seem to be mostly working, thanks!

I did find another example of where it fails and it feels like it shouldn't:
```
t.Run(tc.name, func(t *testing.T) {
			// do things

			if tc.expectedError != nil {
				assert.Error(t, err) <-- testifylint complains here.
			} else {
				require.NoError(t, err)
				if tc.failureDetails != nil {
					assert.Equals(failureDetails, tc.failureDetails)
				}
			}
		}
```

**Comment by Antonboom** [comment]
@jalaziz thank you for involving and making the linter better!

this is a really difficult checker, I'm already thinking about turning it off by default 🥲 

fixed in https://github.com/Antonboom/testifylint/releases/tag/v1.1.2

**Comment by jalaziz** [comment]
> @jalaziz thank you for involving and making the linter better!
> 
> this is a really difficult checker, I'm already thinking about turning it off by default 🥲 
> 
> fixed in https://github.com/Antonboom/testifylint/releases/tag/v1.1.2
> 
> 
> 

Happy to help 🙏🏼. Thank you for the great linter!
